go/enclave/rollupchain/rollup_chain.go

```diff
@@ -705,9 +705,30 @@ func (rc *RollupChain) GetBalance(encryptedParams common.EncryptedParamsGetBalan
 	if blockchainState == nil || err != nil {
 		return nil, err
 	}
-	balance := (*hexutil.Big)(blockchainState.GetBalance(address))
+	balance := (*hexutil.Big)(blockchainState.GetBalance(accountAddress))
 
 	// We encrypt the result.
+	address := accountAddress
+	// If the accountAddress is a contract, encrypt with the address of the contract owner
+	code := blockchainState.GetCode(accountAddress)
+	if len(code) != 0 {
```

Comment:
@joeldudley , I noticed that in `getUserAddrsFromLogTopics`, you're comparing GetCode to nil. 
it should be done like this instead.

Reply:
@tudor-malene Thanks. Did you fix or should I?

Reply:
@joeldudley , I just noticed it. 
Also, would be good to check why it wasn't caught by the tests